### PR TITLE
Fix test_remote_storage_upload_queue_retries flakiness

### LINKED ISSUE #3051: test_remote_storage_upload_queue_retries is flaky
```
2022-12-09T16:04:38.3224170Z =================================== FAILURES ===================================
2022-12-09T16:04:38.3224482Z ______________ test_remote_storage_upload_queue_retries[local_fs] ______________
2022-12-09T16:04:38.3225083Z [gw0] linux -- Python 3.9.2 /github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/bin/python
2022-12-09T16:04:38.3225585Z test_runner/regress/test_remote_storage.py:314: in test_remote_storage_upload_queue_retries
2022-12-09T16:04:38.3225982Z     assert not churn_while_failpoints_active_thread.is_alive()
2022-12-09T16:04:38.3226239Z E   assert not True
2022-12-09T16:04:38.3226656Z E    +  where True = <bound method Thread.is_alive of <Thread(Thread-3, started 140135066326784)>>()
2022-12-09T16:04:38.3227232Z E    +    where <bound method Thread.is_alive of <Thread(Thread-3, started 140135066326784)>> = <Thread(Thread-3, started 140135066326784)>.is_alive
```

https://github.com/neondatabase/neon/actions/runs/3658686712/jobs/6184058965

### PR DESCRIPTION
**Pull request by arpad-m**
* decreases checkpointing and compaction targets for even more layer files
* write 10 thousand rows 2 times instead of writing 20 thousand rows 1 time so that there is more to GC. Before it was noisily jumping between 1 and 0 layer files, now it's jumping between 19 and 20 layer files. The 0 caused an assertion error that gave the test most of its flakiness.
* larger timeout for the churn while failpoints are active thread: this is mostly so that the test is more robust on systems with more load

Fixes #3051

### CODE REVIEW

**Comment by arpad-m** [comment]
The churn while failpoints are active test still complains from time to time. 

There is still a remaining issue which occurs rather rarely:

```
test_runner/regress/test_remote_storage.py:342: in test_remote_storage_upload_queue_retries
    assert not churn_while_failpoints_active_thread.is_alive()
E   assert not True
E    +  where True = <bound method Thread.is_alive of <Thread(Thread-2 (churn_while_failpoints_active), started 139915644892864)>>()
E    +    where <bound method Thread.is_alive of <Thread(Thread-2 (churn_while_failpoints_active), started 139915644892864)>> = <Thread(Thread-2 (churn_while_failpoints_active), started 139915644892864)>.is_alive
```

Basically, even with a larger timeout, we can still run into it. I'm not sure how to fix that one outside of increasing the timeout even further. I have however reduced the number of rows, which should make the test finish faster.

**Comment by koivunej** [comment]
> Basically, even with a larger timeout, we can still run into it. I'm not sure how to fix that one outside of increasing the timeout even further. I have however reduced the number of rows, which should make the test finish faster.

I think this is a rather common problem that we would like to implement the tests to run and fail fast in isolation, but on ci we run a lot of them and some other test could hog all cputime. Because the test order is not random, a new test could had been added recently which started to make this test fail.

However, python `Thread.join` api is so horrible that it is good to keep using it with some timeout. If we would switch this to `join()` then we'd run the pytest level timeout sometimes (or hang -- it doesn't always work). If we would still see the join failing to work then we should raise the value.

File: test_runner/regress/test_remote_storage.py [hunk 1]
```diff
@@ -333,7 +336,7 @@ def churn_while_failpoints_active(result):
 
     # The churn thread doesn't make progress once it blocks on the first wait_completion() call,
     # so, give it some time to wrap up.
-    churn_while_failpoints_active_thread.join(30)
+    churn_while_failpoints_active_thread.join(60)
```
Comment (koivunej) [inline test_runner/regress/test_remote_storage.py:339]: I think we could go straight for 120, or just monitor how it'll go with this.

**Review by koivunej — APPROVED** [review]
Quite unfortunate but I cannot see another one.

File: test_runner/regress/test_remote_storage.py (hunk 1 above)
Comment (arpad-m) [reply test_runner/regress/test_remote_storage.py:339]: the error in here seems to be more rare. let's see how we fare with 60 first.

**Pull request merged**